flow/raster_cache.cc

```diff
@@ -236,6 +265,21 @@ RasterCacheResult RasterCache::Get(Layer* layer, const SkMatrix& ctm) const {
   return it == layer_cache_.end() ? RasterCacheResult() : it->second.image;
 }
 
+RasterCacheResult RasterCache::GetSnapshot(const Layer* layer,
+                                           GrContext* context,
+                                           const SkMatrix& ctm,
```

Comment:
One more thought: the `BackdropFilter` (sent in as `Layer* layer`) owned the immutable `SkImageFilter*`. Hence if this method can guarantee that the filter is coming from the `BackdropFilter`, then the same `layer->unique_id()` implies the same filter.

Reply:
That was the intent - the snapshots are cached by unique_id and so we needed the original layer that did the caching. This could be unraveled by creating another cache indexed by some other unique_id that can be managed separately from the identity of layers.

In my original code I was nulling out the saved filter when it changed, but it doesn't work well because at the framework level we can only tell if the filter is the same object, not if it was a newly constructed version that matches. Thus BDF(filter: ImageFilter.blur(gammas)) would think that you were installing a new filter even if the gammas matched, but if you created a static variable to hold your filter and sent in the same object every time then it would be OK with it.

I think we need to beef up our filtering change detection code either down here or up in framework before this work is done. This is currently an open issue with this fix...